multisig/src/lib.rs

```diff
@@ -47,104 +45,102 @@ impl From<KeyId> for u64 {
-        this
+        let (sk, pk) = secp256k1::generate_keypair(&mut secp256k1::rand::rng());
+        Self {
+            sk: SecretKey { key: sk },
+            pk: PublicKey { key: pk },
+        }
     }
 
     /// Generate keypair from a seed.
     pub fn from_seed(seed: [u8; 32]) -> Self {
-        let this = Self {
-            pair: ed25519::KeyPair::from_seed(ed25519::Seed::new(seed)),
-        };
-        assert!(!SMALL_ORDER_KEYS.contains(&&*this.pair.pk));
-        this
+        loop {
+            if let Ok(sk) = secp256k1::SecretKey::from_byte_array(seed) {
+                let pk = sk.public_key(secp256k1::SECP256K1);
+                return Self {
+                    sk: SecretKey { key: sk },
+                    pk: PublicKey { key: pk },
+                };
+            }
+        }
     }
 
     /// Returns ed25519 Public key.
     pub fn public_key(&self) -> PublicKey {
-        PublicKey { key: self.pair.pk }
+        self.pk
     }
 
     /// Return ed25519 secret key.
     pub fn secret_key(&self) -> SecretKey {
-        SecretKey {
-            key: self.pair.sk.clone(),
-        }
+        self.sk.clone()
     }
 
     /// Sign data with our ed25519 secret key.
-    pub fn sign(&self, data: &[u8], deterministic: bool) -> Signature {
-        Signature {
-            sig: self
-                .pair
-                .sk
-                .sign(data, (!deterministic).then(ed25519::Noise::generate)),
-        }
+    pub fn sign(&self, data: &[u8]) -> Signature {
+        self.sk.sign(data)
     }
 }
 
 impl PublicKey {
     pub fn is_valid(&self, data: &[u8], s: &Signature) -> bool {
-        self.key.verify(data, &s.sig).is_ok()
+        use secp256k1::hashes::{Hash, sha256};
+        let hash = sha256::Hash::hash(data);
+        let mesg = secp256k1::Message::from_digest(hash.to_byte_array());
+        s.sig.verify(mesg, &self.key).is_ok()
     }
 
-    pub fn as_bytes(&self) -> [u8; 32] {
-        *self.key
-    }
-
-    pub fn as_slice(&self) -> &[u8] {
-        &self.key[..]
+    pub fn to_bytes(&self) -> [u8; 33] {
+        self.key.serialize()
     }
 }
 
 impl SecretKey {
-    pub fn sign(&self, data: &[u8], deterministic: bool) -> Signature {
+    pub fn sign(&self, data: &[u8]) -> Signature {
+        use secp256k1::hashes::{Hash, sha256};
+        let hash = sha256::Hash::hash(data);
+        let mesg = secp256k1::Message::from_digest(hash.to_byte_array());
         Signature {
-            sig: self
-                .key
-                .sign(data, (!deterministic).then(ed25519::Noise::generate)),
+            sig: self.key.sign_ecdsa(mesg),
         }
     }
 
     pub fn public_key(&self) -> PublicKey {
-        PublicKey {
-            key: self.key.public_key(),
-        }
+        let pk = self.key.public_key(secp256k1::SECP256K1);
+        PublicKey { key: pk }
     }
 
-    pub fn as_bytes(&self) -> [u8; 64] {
-        *self.key
+    pub fn as_bytes(&self) -> [u8; 32] {
```

Comment:
there are many cases of `as_bytes` and `to_bytes` i don't know the convention we are following.

Reply:
We should follow these https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv